src/nat/finetuning/interfaces/finetuning_runner.py

```
@@ -0,0 +1,261 @@
+            # Run evaluation
+            eval_output = await validation_builder.run_eval()
+
+            # Calculate validation metrics from eval output
+            validation_metrics = self._calculate_validation_metrics(eval_output)
+            validation_metrics["epoch"] = epoch
+            validation_metrics["dataset_type"] = "validation"
+
+            logger.info("Validation metrics for epoch %d: %s", epoch, validation_metrics)
+            return validation_metrics
+
+        except Exception as e:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)